gauntlet/packages/gauntlet-solana-contracts/src/commands/contracts/ocr2/proposal/acceptProposal.ts

```diff
@@ -1,47 +1,158 @@
-    // const oracles = aggregatorOperators.map((operator) => ({
-    //   // Same here
-    //   transmitter: operator.ocrNodeAddress[0],
-    //   signer: operator.ocr2OnchainPublicKey[0].replace('ocr2on_solana_', ''),
-    // }))
-    const digest = Buffer.alloc(32) // TODO
 
+    const rdd = getRDD(this.flags.rdd)
+    const aggregator = rdd.contracts[this.flags.state]
+    const _toHex = (a: string) => Buffer.from(a, 'hex')
+    const aggregatorOperators: any[] = aggregator.oracles.map((o) => rdd.operators[o.operator])
+    const oracles = aggregatorOperators
+      .map((operator) => ({
+        transmitter: operator.ocrNodeAddress[0],
+        signer: operator.ocr2OnchainPublicKey[0].replace('ocr2on_solana_', ''),
+        payee: operator.adminAddress,
+      }))
+      .sort((a, b) => Buffer.compare(_toHex(a.signer), _toHex(b.signer)))
+    const offchainConfig = ProposeOffchainConfig.makeInputFromRDD(rdd, this.flags.state)
+    const tokenMint = process.env.LINK!
+    const f = aggregator.config.f
     return {
-      digest,
+      version: 2,
+      f,
+      tokenMint,
+      oracles,
+      offchainConfig,
+      randomSecret: this.flags.secret,
     }
   }
 
   constructor(flags, args) {
     super(flags, args)
 
     this.require(!!this.flags.state, 'Please provide flags with "state"')
-    this.require(!!this.flags.proposal, 'Please provide flags with "proposal"')
+    this.require(!!this.flags.proposalId, 'Please provide flags with "proposal"')
+    this.require(!!this.flags.secret, 'Please provide flags with "secret"')
+    this.require(!!process.env.LINK, 'Please set the LINK env var')
+    this.require(!!process.env.SECRET, 'Please set the SECRET env var')
+  }
+
+  makeDigestInputFromProposal = (proposalInfo: Proposal): DigestInput => {
+    const oracles = proposalInfo.oracles.xs
+      .map((oracle) => {
+        return {
+          transmitter: new PublicKey(oracle.transmitter),
+          signer: Buffer.from(oracle.signer.key),
+          payee: new PublicKey(oracle.payee),
+        }
+      })
+      .slice(0, proposalInfo.oracles.len)
+    return {
+      version: new BN(proposalInfo.offchainConfig.version),
+      f: new BN(proposalInfo.f),
+      tokenMint: new PublicKey(proposalInfo.tokenMint),
+      oracles,
+      offchainConfig: proposalInfo.offchainConfig.xs.slice(0, proposalInfo.offchainConfig.len),
+    }
+  }
+
+  makeDigestInput = async (input: Input): Promise<DigestInput> => {
+    return {
+      version: new BN(2),
+      f: new BN(input.f),
+      tokenMint: new PublicKey(input.tokenMint),
+      oracles: input.oracles.map((oracle) => {
+        return {
+          transmitter: new PublicKey(oracle.transmitter),
+          signer: Buffer.from(oracle.signer, 'hex'),
+          payee: new PublicKey(oracle.payee),
+        }
+      }),
+      offchainConfig: await (
+        await serializeOffchainConfig(input.offchainConfig, process.env.SECRET!, input.randomSecret)
+      ).offchainConfig,
+    }
+  }
```

Comment:
I think rather than generating from `RDD` data, this command could take a hex `digest` input provided by the user.